controller/config_test.go

```diff
@@ -121,6 +121,20 @@ var validateTests = []struct {
 		controller.CACertKey:         testing.CACert,
 	},
 	expectError: `invalid identity public key: wrong length for base64 key, got 3 want 32`,
+}, {
+	about: "invalid HA space name",
+	config: controller.Config{
+		controller.CACertKey:   testing.CACert,
+		controller.JujuHASpace: `Ǣ`,
```

Comment:
we should also have a test with:

controller:JujuHASpace: 1,

and similar (eg:
controller.JujuManagementSpace: true,
)

generally just things that aren't strings.
I'm not a huge fan of using Unicode, but its fine. (mostly because editors don't always assume utf-8 and then you get mojibake if someone does the wrong thing in latin-1, etc.)

Reply:
@jameinel The way that validation is coded right now is that it takes the supplied value and casts it into a string. This means that both `1` and `true` are valid space names. We have alternatively provided other test data to show working validation.